wsmaster/codenvy-hosted-api-resource/src/main/java/com/codenvy/resource/api/free/FreeResourcesManager.java

```diff
@@ -0,0 +1,128 @@
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Facade for free resources limit related operations.
+ *
+ * @author Sergii Leschenko
+ */
+@Singleton
+public class FreeResourcesManager {
+    private final FreeResourcesLimitDao   freeResourcesLimitDao;
+    private final ResourceInstanceCreator resourceInstanceCreator;
+
+    @Inject
+    public FreeResourcesManager(FreeResourcesLimitDao freeResourcesLimitDao,
+                                ResourceInstanceCreator resourceInstanceCreator) {
+        this.freeResourcesLimitDao = freeResourcesLimitDao;
+        this.resourceInstanceCreator = resourceInstanceCreator;
+    }
+
+    /**
+     * Stores (creates new one or updates existed) free resource limit.
+     *
+     * @param freeResourcesLimit
+     *         resources limit to store
+     * @return stored resources limit
+     * @throws NullPointerException
+     *         when {@code freeResourcesLimit} is null
+     * @throws NotFoundException
+     *         when resources limit contains resource with non supported type
+     * @throws ServerException
+     *         when any other error occurs
+     */
+    public FreeResourcesLimit store(FreeResourcesLimit freeResourcesLimit) throws NotFoundException,
+                                                                                  ServerException {
+        requireNonNull(freeResourcesLimit, "Required non-null free resources limit");
+        List<AbstractResource> resources = new ArrayList<>();
+        for (Resource resource : freeResourcesLimit.getResources()) {
+            resources.add(resourceInstanceCreator.newInstance(resource));
+        }
+
+        final FreeResourcesLimitImpl toStore = new FreeResourcesLimitImpl(freeResourcesLimit.getAccountId(),
+                                                                          resources);
+        freeResourcesLimitDao.store(toStore);
+        return toStore;
+    }
+
+    /**
+     * Returns free resources limit for account with specified id
+     *
+     * @param accountId
+     *         account id to fetch resources limit
+     * @return free resources limit for account with specified id
+     * @throws NullPointerException
+     *         when {@code accountId} is null
+     * @throws NotFoundException
+     *         when free resources limit for specifies id was not found
+     * @throws ServerException
+     *         when any other error occurs
+     */
+    public FreeResourcesLimit get(String accountId) throws NotFoundException, ServerException {
+        requireNonNull(accountId, "Required non-null account id");
+        return freeResourcesLimitDao.get(accountId);
+    }
+
+    /**
+     * Removes free resources limit for account with specified id
+     *
+     * Note: after removing resources limit account will be able to use default resources
```

Comment:
Consider adding `<p>` before the Note, generated javadoc will be like this `specified idNote:`